t/semaphore.t

```diff
@@ -0,0 +1,1068 @@
+                error("create failed")
+            end
+            local function func(sem)
+                sem:wait(10)
+            end
+            local co = ngx.thread.spawn(func,sem)
+            ngx.thread.kill(co)
+            ngx.say("test")
+       ';
+    }
+--- log_level: debug
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+test
+--- error_log
+ngx_http_lua_semaphore_cleanup
+
+
+
+=== TEST 19: a light thread that is going to exit is waitting a semaphore
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+       content_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            if not sem then
+                error("create failed")
+            end
+            local function func(sem)
+                sem:wait(10)
+            end
+            local co = ngx.thread.spawn(func,sem)
+            ngx.say("test")
+            ngx.exit(200)
+       ';
+    }
+--- log_level: debug
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+test
+--- error_log
+ngx_http_lua_semaphore_cleanup
+
+
+
+=== TEST 20: test thread wait or post a sem in one request
+--- http_config eval: $::HttpConfig
+--- config
+    location = /test {
+        access_log off;
+        content_by_lua '
+            local semaphore = require "ngx.semaphore"
+
+            local g = getmetatable(_G).__index
+            local sem, err = semaphore.new(0)
+            if not sem then
+                ngx.say(err)
+                ngx.exit(500)
+            end
+            g.test = sem
+
+            local sem2, err = semaphore.new(0)
+            if not sem2 then
+                ngx.say(err)
+                ngx.exit(500)
+            end
+            g.test_next = sem2
+
+            g.log = {}
+            g.log_i = 0
+            g.wait_id = 0
+            g.wakeup_id = 0
+            --0:wait 1:post 2:post_all
+            g.test_op = {0,1,1,0}
+            local function func(op,index,g)
+                local res
+                if op == 0 then
+                    local sem = g.test
+                    local sem2 = g.test_next
+                    sem2:post()
+                    g.wait_id = g.wait_id +1
+                    local wait_id = g.wait_id
+                    local ok, err = sem:wait(10)
+                    g.wakeup_id = g.wakeup_id + 1
+                    g.log_i = g.log_i + 1
+                    local wake_id = g.wakeup_id
+                    g.log[g.log_i] = {state=ok, err=err,op=0,wait_id = wait_id,wake_id = wake_id}
+                else
+                    local sem = g.test
+                    local sem2 = g.test_next
+                    sem2:post()
+                    local ok, err = sem:post()
+                    g.log_i = g.log_i + 1
+                    g.log[g.log_i] = {state=ok, err=err,op=1}
+                end
+            end
+            local co_array = {}
+            for i=1,#g.test_op do
```

Comment:
Style: we need spaces around binary operators like `=`. We also need a space after `,`. Please fix other places with similar issues.